ddht/v5_1/alexandria/partials/_utils.py

```diff
@@ -47,3 +48,15 @@ def get_chunk_count_for_data_length(length: int) -> int:
     if length == 0:
         return 0
     return (length + CHUNK_SIZE - 1) // CHUNK_SIZE  # type: ignore
+
+
+@to_tuple
+def merge_paths(*paths: TreePath) -> Iterable[TreePath]:
```

Comment:
A docstring could help here, I'm not sure what this function is doing. It takes a bunch of paths and de-duplicates them? If any path is a prefix of another the shorter path is dropped. From the name it sounded like it would somehow combine paths to make other paths, if I've got what it's doing right maybe this could be called `deduplicate_paths`.

Reply:
I think it's something like `remove_overlapped_paths` as it removes any path that is on another longer path.